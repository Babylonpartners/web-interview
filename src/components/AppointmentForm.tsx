import React from 'react'
import '../styles/App.scss'
import { ConsultantTypes } from './ConsultantTypes'
import { AppointmentTimes } from './AppointmentTimes'
import { AppointmentTypes } from './AppointmentTypes'
import { AppointmentNotes } from './AppointmentNotes'
import { Responsive } from 'semantic-ui-react'
import { SubmitButton } from './SubmitButton'

interface Props {
  availableSlots: string[],
  availableAppTypes: string[],
  availableSlotsLoading: boolean,
  user: any,
  userLoading: boolean,

  selectedConsultantType: string,
  selectedAppointmentTime: string,
  selectedAppointmentType: string,
  appointmentNotes: string,

  selectedConsultantTypeError: string,
  selectedAppointmentTimeError: string,
  selectedAppointmentTypeError: string,
  appointmentNotesError: string,

  handleSubmit: any,
  handleAppointmentNotesChange: any,
  handleAppointmentTimesChange: any,
  handleAppointmentTypesChange: any,
  handleConsultantTypesChange: any,
}

interface State {
  
}

export default class AppointmentForm extends React.PureComponent<Props, State> {
 

  render() {


    return (

      <>

        <Responsive maxWidth={640}>
        <div style={{ maxWidth: 600, margin: '10px auto' }}>
            <form 
              data-testid="appointmentForm" 
              onSubmit={this.props.handleSubmit}
            >
            <ConsultantTypes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleConsultantTypesChange}
                selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentTimes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleAppointmentTimesChange}
                selectedConsultantType={this.props.selectedConsultantType}
                selectedAppointmentTime={this.props.selectedAppointmentTime}
            /> 
            <br />
            <br />
            <AppointmentTypes
              availableSlots={this.props.availableSlots}
              handleChange={this.props.handleAppointmentTypesChange}
              selectedAppointmentType={this.props.selectedAppointmentType}
              selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentNotes 
              handleChange={this.props.handleAppointmentNotesChange}
              appointmentNotesError={this.props.appointmentNotesError}
            />
            <SubmitButton />
            </form>
          </div>
        </Responsive>

        <Responsive minWidth={641} maxWidth={768}>
        <div style={{ maxWidth: 600, margin: '24px auto' }}>
            <form 
              data-testid="appointmentForm" 
              onSubmit={this.props.handleSubmit}
            >
            <ConsultantTypes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleConsultantTypesChange}
                selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentTimes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleAppointmentTimesChange}
                selectedConsultantType={this.props.selectedConsultantType}
                selectedAppointmentTime={this.props.selectedAppointmentTime}
            /> 
            <br />
            <br />
            <AppointmentTypes
              availableSlots={this.props.availableSlots}
              handleChange={this.props.handleAppointmentTypesChange}
              selectedAppointmentType={this.props.selectedAppointmentType}
              selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentNotes 
              handleChange={this.props.handleAppointmentNotesChange}
              appointmentNotesError={this.props.appointmentNotesError}
            />
           <SubmitButton />
            </form>
          </div>
        </Responsive>

        <Responsive minWidth={769} maxWidth={992}>
        <div style={{ maxWidth: 800, margin: '24px auto' }}>
            <form 
              data-testid="appointmentForm" 
              onSubmit={this.props.handleSubmit}
            >
            <ConsultantTypes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleConsultantTypesChange}
                selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentTimes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleAppointmentTimesChange}
                selectedConsultantType={this.props.selectedConsultantType}
                selectedAppointmentTime={this.props.selectedAppointmentTime}
            /> 
            <br />
            <br />
            <AppointmentTypes
              availableSlots={this.props.availableSlots}
              handleChange={this.props.handleAppointmentTypesChange}
              selectedAppointmentType={this.props.selectedAppointmentType}
              selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentNotes 
              handleChange={this.props.handleAppointmentNotesChange}
              appointmentNotesError={this.props.appointmentNotesError}
            />
            <SubmitButton />
            </form>
          </div>
        </Responsive>
      
        <Responsive minWidth={993} maxWidth={1200}>
        <div style={{ maxWidth: 900, margin: '24px auto' }}>
            <form 
              data-testid="appointmentForm" 
              onSubmit={this.props.handleSubmit}
            >
            <ConsultantTypes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleConsultantTypesChange}
                selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentTimes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleAppointmentTimesChange}
                selectedConsultantType={this.props.selectedConsultantType}
                selectedAppointmentTime={this.props.selectedAppointmentTime}
            /> 
            <br />
            <br />
            <AppointmentTypes
              availableSlots={this.props.availableSlots}
              handleChange={this.props.handleAppointmentTypesChange}
              selectedAppointmentType={this.props.selectedAppointmentType}
              selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentNotes 
              handleChange={this.props.handleAppointmentNotesChange}
              appointmentNotesError={this.props.appointmentNotesError}
            />
            <SubmitButton />
            </form>
          </div>
        </Responsive>

        <Responsive minWidth={1201}>
        <div style={{ maxWidth: 1000, margin: '24px auto' }}>
            <form 
              data-testid="appointmentForm" 
              onSubmit={this.props.handleSubmit}
            >
            <ConsultantTypes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleConsultantTypesChange}
                selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentTimes
                availableSlots={this.props.availableSlots}
                handleChange={this.props.handleAppointmentTimesChange}
                selectedConsultantType={this.props.selectedConsultantType}
                selectedAppointmentTime={this.props.selectedAppointmentTime}
            /> 
            <br />
            <br />
            <AppointmentTypes
              availableSlots={this.props.availableSlots}
              handleChange={this.props.handleAppointmentTypesChange}
              selectedAppointmentType={this.props.selectedAppointmentType}
              selectedConsultantType={this.props.selectedConsultantType}
            />
            <br />
            <br />
            <AppointmentNotes 
              handleChange={this.props.handleAppointmentNotesChange}
              appointmentNotesError={this.props.appointmentNotesError}
            />
            <SubmitButton />
            </form>
          </div>
        </Responsive>
       
      </>
    )
  }
}
