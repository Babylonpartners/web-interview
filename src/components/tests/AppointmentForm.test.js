import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import AppointmentForm from '../AppointmentForm'

afterEach(() => {
  cleanup()
})

const submit = jest.fn()

test('<AppointmentForm>', async () => {
  const { getByTestId, queryByTestId, container } = render(
    <AppointmentForm
      availableSlots
      availableAppTypes
      availableSlotsLoading
      user
      handleSubmit={submit}
    />
  )
  expect(container.firstChild).toMatchSnapshot()

  expect(queryByTestId('appointmentForm')).toBeTruthy()
  expect(queryByTestId('appointmentNotesComponent')).toBeTruthy()

  queryByTestId('appointmentNotesTextArea').value = ''
  fireEvent.change(queryByTestId('appointmentNotesTextArea'))

  fireEvent.submit(getByTestId('appointmentForm'))
  expect(submit).toHaveBeenCalled()
  expect(submit).toHaveBeenCalledTimes(1)

  // expect(submit).toHaveBeenCalledWith({
  //   selectedConsultantType: 'gp',
  //   selectedAppointmentTime: '',
  //   selectedAppointmentType: '',
  //   appointmentNotes: '',
  // })
})
