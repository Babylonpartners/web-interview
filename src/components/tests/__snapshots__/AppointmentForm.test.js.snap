// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentForm> 1`] = `
<div>
  <div
    style="max-width: 900px; margin: 24px auto;"
  >
    <form
      data-testid="appointmentForm"
    >
      <p
        data-testid="consultantTypesLoading"
      >
        Loading...
      </p>
      <br />
      <br />
      <p
        data-testid="appointmentTimesLoading"
      >
        Loading...
      </p>
      <br />
      <br />
      <p
        data-testid="appointmentTypesLoading"
      >
        Loading...
      </p>
      <br />
      <br />
      <div
        data-testid="appointmentNotesComponent"
      >
        <strong
          data-testid="appointmentNotesHeader"
          style="color: rgb(72, 72, 72); font-family: 'Roboto', sans-serif; margin-bottom: 5px;"
        >
          Notes
        </strong>
        <br />
        <textarea
          class="notes"
          data-testid="appointmentNotesTextArea"
          placeholder="Describe your symptoms"
          style="color: rgb(72, 72, 72); font-family: 'Roboto', sans-serif; margin-bottom: 5px;"
        />
        <br />
        <br />
        <br />
        <br />
        <br />
      </div>
      <hr />
      <br />
      <div>
        <button
          class="submit-button"
          data-testid="submitButton"
          style="font-family: 'Roboto', sans-serif; margin-bottom: 5px;"
          type="submit"
        >
          Book
        </button>
        <br />
        <br />
        <br />
      </div>
    </form>
  </div>
</div>
`;
